cdc/puller/puller.go

```diff
@@ -198,48 +198,50 @@ func (p *pullerImpl) Run(ctx context.Context) error {
 			}
 
 			if e.Resolved != nil {
-				metricTxnCollectCounterResolved.Inc()
-				if !regionspan.IsSubSpan(e.Resolved.Span, p.spans...) {
-					log.Panic("the resolved span is not in the total span",
-						zap.String("namespace", p.changefeed.Namespace),
-						zap.String("changefeed", p.changefeed.ID),
-						zap.Int64("tableID", p.tableID),
-						zap.String("tableName", p.tableName),
-						zap.Any("resolved", e.Resolved),
-						zap.Any("spans", p.spans),
-					)
-				}
-				// Forward is called in a single thread
-				p.tsTracker.Forward(e.Resolved.Span, e.Resolved.ResolvedTs)
-				resolvedTs := p.tsTracker.Frontier()
-				if resolvedTs > 0 && !initialized {
-					// Advancing to a non-zero value means the puller level
-					// resolved ts is initialized.
-					atomic.StoreInt64(&p.initialized, 1)
-					initialized = true
+				for _, resolvedSpan := range e.Resolved {
+					metricTxnCollectCounterResolved.Inc()
+					if !regionspan.IsSubSpan(resolvedSpan.Span, p.spans...) {
+						log.Panic("the resolved span is not in the total span",
+							zap.String("namespace", p.changefeed.Namespace),
+							zap.String("changefeed", p.changefeed.ID),
+							zap.Int64("tableID", p.tableID),
+							zap.String("tableName", p.tableName),
+							zap.Any("resolved", e.Resolved),
+							zap.Any("spans", p.spans),
+						)
+					}
+					// Forward is called in a single thread
+					p.tsTracker.Forward(resolvedSpan.Span, resolvedSpan.ResolvedTs)
+					resolvedTs := p.tsTracker.Frontier()
+					if resolvedTs > 0 && !initialized {
+						// Advancing to a non-zero value means the puller level
+						// resolved ts is initialized.
+						atomic.StoreInt64(&p.initialized, 1)
```

Comment:
`Puller.IsInitialized()` is only used in the test now, we can remove it

Reply:
I will remove it in next PR